var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -26,18 +29,23 @@ class PyH5py(PythonPackage):
 
     variant('mpi', default=True, description='Build with MPI support')
 
+    # Python versions
+    depends_on('python@3.6:', when='@3.0.0:')
```

Comment:
What type does it need to be? build/run? build/link/run?

Reply:
There's no python linking from the final shared library object so I've described it as build/run. Is that the way to determine the dependency type ? 

```
[sajid@xrmlite ~]$ ldd /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/py-h5py-3.1.0-ew2jwuk2klt43qlhyl6orpjpftdeg4th/lib/python3.8/site-packages/h5py/h5.cpython-38-x86_64-linux-gnu.so
        linux-vdso.so.1 (0x00007ffc8c75a000)
        libhdf5.so.103 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/hdf5-1.10.7-v7dc6lvchqloxbjxyby6ukmarkbh7zok/lib/libhdf5.so.103 (0x000014ac96e29000)
        libhdf5_hl.so.100 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/hdf5-1.10.7-v7dc6lvchqloxbjxyby6ukmarkbh7zok/lib/libhdf5_hl.so.100 (0x000014ac96c04000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x000014ac969e4000)
        libc.so.6 => /lib64/libc.so.6 (0x000014ac96621000)
        libz.so.1 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/zlib-1.2.11-mdobuge7nvmkbbyz3ysypyxktl3zmdz5/lib/libz.so.1 (0x000014ac9640a000)
        libdl.so.2 => /lib64/libdl.so.2 (0x000014ac96206000)
        libm.so.6 => /lib64/libm.so.6 (0x000014ac95e84000)
        libmpi.so.40 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/openmpi-3.1.6-qv64jrpun3r2pdv4frhdf5ckhbffnzd4/lib/libmpi.so.40 (0x000014ac95a22000)
        /lib64/ld-linux-x86-64.so.2 (0x000014ac9761f000)
        libopen-rte.so.40 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/openmpi-3.1.6-qv64jrpun3r2pdv4frhdf5ckhbffnzd4/lib/libopen-rte.so.40 (0x000014ac956f3000)
        libopen-pal.so.40 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/openmpi-3.1.6-qv64jrpun3r2pdv4frhdf5ckhbffnzd4/lib/libopen-pal.so.40 (0x000014ac952d9000)
        librt.so.1 => /lib64/librt.so.1 (0x000014ac950d0000)
        libutil.so.1 => /lib64/libutil.so.1 (0x000014ac94ecc000)
        libhwloc.so.5 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/hwloc-1.11.11-iomswist7ntzr3rpm4kq5kt7esg5r3ww/lib/libhwloc.so.5 (0x000014ac94c8d000)
        libnuma.so.1 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/numactl-2.0.14-qzfclb5dy7exe5xlis5ckgxucrhow5r2/lib/libnuma.so.1 (0x000014ac94a82000)
        libpciaccess.so.0 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/libpciaccess-0.16-fcifgo6fdwqo5ipgrm2kjlj3udqj4sqt/lib/libpciaccess.so.0 (0x000014ac94879000)
        libxml2.so.2 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/libxml2-2.9.10-pwkeovrz4p5hguljx4xxris5a2hxag7d/lib/libxml2.so.2 (0x000014ac94516000)
        liblzma.so.5 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/xz-5.2.5-q5425k625lweozuobrlmohln7gjqrbpq/lib/liblzma.so.5 (0x000014ac942f0000)
        libiconv.so.2 => /home/sajid/packages/spack/opt/spack/linux-centos8-broadwell/gcc-9.3.0/libiconv-1.16-rh6ayrzpbdmeuxq4d22xcgbzzg7o4w3e/lib/libiconv.so.2 (0x000014ac93ff3000)
[sajid@xrmlite ~]$
```

Reply:
Either check for linking like you did, or check the build log. Sometimes you'll see that it links to the header files but not the library.